From 9b07a5aa6eb3f7315dce3436e2f8c9ff2d9fd41a Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Tue, 16 Dec 2014 15:22:49 +0530
Subject: [PATCH 1/1] Dollar Cove ADC: This change modifies the ADC setting for
 enabling SYSTHERM.

The existing gpadc setting does not enable the SYSTHERM sampling.
The change enables the GPADC setting which sets the GPIO0 pin to output
current used for SYSTHERM ADC sampling.

Change-Id: I7de81182a886defe8d41a6bb77cf4d237d992d9e
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
---
 drivers/iio/adc/dc_xpwr_gpadc.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/adc/dc_xpwr_gpadc.c b/drivers/iio/adc/dc_xpwr_gpadc.c
index 07442bd..5185008 100644
--- a/drivers/iio/adc/dc_xpwr_gpadc.c
+++ b/drivers/iio/adc/dc_xpwr_gpadc.c
@@ -48,7 +48,10 @@
 #define ADC_EN_MASK			0xF1
 
 #define DC_PMIC_ADC_CNTL_REG		0x84
+
 #define DC_PMIC_TSP_CNTL_REG		0x85
+#define ADC_GPIO0_OUTPUT_CURRENT	0xB4
+
 #define ADC_PMIC_TEMP_DATAH_REG		0x56
 #define ADC_PMIC_TEMP_DATAL_REG		0x57
 #define ADC_TSP_DATAH_REG		0x58
@@ -73,7 +76,8 @@
 #define ADC_NON_BAT_CUR_DATAL_MASK	0x0F
 
 #define ADC_TS_PIN_CNRTL_REG           0x84
-#define ADC_TS_PIN_ON                  0xF3
+#define ADC_TS_PIN_ON                  0xF2
+
 
 #define DEV_NAME			"dollar_cove_adc"
 
@@ -204,7 +208,7 @@ static int dc_xpwr_gpadc_probe(struct platform_device *pdev)
 	platform_set_drvdata(pdev, indio_dev);
 	mutex_init(&info->lock);
 
-	/* Current Source from TS pin always ON */
+	/* Use TS current source */
 	intel_soc_pmic_writeb(ADC_TS_PIN_CNRTL_REG, ADC_TS_PIN_ON);
 
 	/*
@@ -214,6 +218,9 @@ static int dc_xpwr_gpadc_probe(struct platform_device *pdev)
 	 */
 	intel_soc_pmic_writeb(DC_PMIC_ADC_EN_REG, ADC_EN_MASK);
 
+	/* Set the GPIO0 ADC output current */
+	intel_soc_pmic_writeb(DC_PMIC_TSP_CNTL_REG, ADC_GPIO0_OUTPUT_CURRENT);
+
 	indio_dev->dev.parent = &pdev->dev;
 	indio_dev->name = pdev->name;
 	indio_dev->channels = dc_xpwr_adc_channels;
-- 
1.9.1

